crates/matrix-sdk/tests/integration/encryption/backups.rs

```diff
@@ -478,6 +498,306 @@ async fn steady_state_waiting() {
+        )),
+    )
+    .await;
+
+    setup_create_room_and_send_message_mocks(&server).await;
+
+    backups.create().await.expect("We should be able to create a new backup");
+
+    let invite = vec![];
+    let request = assign!(CreateRoomRequest::new(), {
+        invite,
+        is_direct: true,
+    });
+
+    let alice_room = client.create_room(request).await?;
+
+    alice_room.enable_encryption().await?;
+
+    assert!(alice_room.is_encrypted().await?, "room should be encrypted");
+
+    // Send a message to create an outbound session that should be uploaded to
+    // backup
+    let content = RoomMessageEventContent::text_plain("Hello world");
+    let txn_id = TransactionId::new();
+    let _ = alice_room.send(content).with_transaction_id(&txn_id).await?;
+
+    just_mount(
+        &server,
+        "GET",
+        "/_matrix/client/r0/sync",
+        ResponseTemplate::new(200).set_body_json(json!({
+            "next_batch": "sfooBar",
+        "device_one_time_keys_count": {
+            "signed_curve25519": 50
+        },
+        "org.matrix.msc2732.device_unused_fallback_key_types": [
+            "signed_curve25519"
+        ],
+        "device_unused_fallback_key_types": [
+            "signed_curve25519"
+        ]
+        })),
+    )
+    .await;
+
+    client.sync_once(Default::default()).await?;
+
+    server.verify().await;
+    Ok(())
+}
+
+#[async_test]
+#[cfg(feature = "experimental-sliding-sync")]
+async fn incremental_upload_of_keys_sliding_sync() -> Result<()> {
+    let user_id = user_id!("@example:morpheus.localhost");
+
+    let session = MatrixSession {
+        meta: SessionMeta { user_id: user_id.into(), device_id: device_id!("DEVICEID").to_owned() },
+        tokens: MatrixSessionTokens { access_token: "1234".to_owned(), refresh_token: None },
+    };
+    let server = wiremock::MockServer::start().await;
+    let builder = Client::builder()
+        .homeserver_url(server.uri())
+        .sliding_sync_proxy(server.uri())
+        .server_versions([ruma::api::MatrixVersion::V1_0]);
+
+    let client =
+        builder.request_config(RequestConfig::new().disable_retry()).build().await.unwrap();
+
+    client.restore_session(session).await.unwrap();
+
+    let backups = client.encryption().backups();
+
+    // This is the call we want to check. The newly created outbound session should
+    // be uploaded to backup.
+    mount_once(
+        &server,
+        "PUT",
+        "_matrix/client/unstable/room_keys/keys",
+        ResponseTemplate::new(200).set_body_json(json!({
+            "count": 1,
+            "etag": "abcdefg",
+        }
+        )),
+    )
+    .await;
+
+    setup_create_room_and_send_message_mocks(&server).await;
+
+    backups.create().await.expect("We should be able to create a new backup");
+
+    let invite = vec![];
+    let request = assign!(CreateRoomRequest::new(), {
+        invite,
+        is_direct: true,
+    });
+
+    let alice_room = client.create_room(request).await?;
```

Comment:
nit: can you inline `invite` and `request` into the `create_room` call? (fewer variables => lower cognitive complexity)

Reply:
I tried, but don't see how to make it work. Looks that there is a problem with the assign! macro. FWIW I just copy pasted this code from other tests